CMSgov/cms-carts-seds
# Add stack termination protection to bigmac streams

### Description
Stack termination protection is a [serverless plugin](https://www.serverless.com/plugins/serverless-stack-termination-protection) to use an [AWS feature](https://docs.aws.amazon.com/emr/latest/ManagementGuide/UsingEMR_TerminationProtection.html) that prevents long-running clusters from being accidentally terminated. They still can be, but you have to remove the protection first.

Across all apps managed by our team (SEDS, CARTS, QMR, MCR, MFP), there are 37 `serverless.yml` files for defining AWS stacks. Of those, 36 already include `serverless-stack-termination-protection`. `carts-bigmac-streams` is the 37th. If termination protection is a good idea for all our other stacks, I think we want it for this one as well.

### Related ticket(s)
n/a

---
### How to test
1. Log in to AWS
2. Follow the steps described [here](https://docs.aws.amazon.com/emr/latest/ManagementGuide/UsingEMR_TerminationProtection.html#emr-termination-protection-running-cluster) to confirm that protection is enabled

### Important updates
n/a

---
### Author checklist
